src/Standards/Squiz/Docs/Classes/SelfMemberReferenceStandard.xml

```diff
@@ -1,30 +1,56 @@
 <documentation title="Self Member Reference">
     <standard>
     <![CDATA[
-    The self keyword should be used instead of the current class name, should be lowercase, and should not have spaces before or after it.
+    When referencing static members within a class, the self keyword must be used instead of the
+    current class name, it must be lowercase, and there must be no spaces before or after the double
+    colon.
```

Comment:
This rephrasing is confusing to me.

"static members within a class" will be interpreted as methods/properties declared with the `static` keyword, while that's not something the sniff takes into account.

The `static` in the _"local static member reference"_ in the error messages is about _static references_, not _static members_.

Reply:
> calling a non-static member with a static reference causes a warning or fatal error depending on the PHP version

Actually, as this sniff is about hierarchy keywords, IIRC this won't be a warning nor a fatal error in any PHP version.

> Or do you prefer the original approach

I think I do as it makes it more straight-forward to understand what the sniff expects, though the description does contain multiple rules in one description.

Might be worth splitting up the description to have a separate description for each code sample ?

Reply:
> Actually, as this sniff is about hierarchy keywords, IIRC this won't be a warning nor a fatal error in any PHP version.

Perhaps I'm still missing something, but here is a code example that illustrates what I have in mind. This example is a fatal error in PHP >= 8.0, but it works with a notice in older versions. It is an example of calling a non-static member with a static reference, and it does trigger the sniff.

https://3v4l.org/gRAUG

> Might be worth splitting up the description to have a separate description for each code sample ?

I went ahead and created a `<standard>` block for each of the `<code_comparison>` blocks.

Reply:
> I see what you mean and now understand the confusion.
> I was talking about this: https://3v4l.org/160e6

Ah, I see, thanks for the code sample.